### PR TITLE
Migrate the project to use solid_lints

### PR DESCRIPTION
**Pull request by solid-yuriiprykhodko**

### CODE REVIEW

File: test/hydrated_test.dart [hunk 1]
```diff
@@ -169,40 +201,38 @@ void main() {
...
+        };
+        final s = HydratedSubject<int>(
+          key,
+          keyValueStore: mockKeyValueStore,
         );
+
+        await pumpEventQueue();
+
+        expect(s.error, isA<Exception>());
       });
```
Comment (solid-yuriiprykhodko) [inline test/hydrated_test.dart:218]: This change comes from the fact that now we propagate all exceptions to the `HydratedSubject` Stream.

This gives user the freedom of choice when it comes to catching Errors.

File: lib/src/key_value_store/shared_preferences_store.dart [hunk 2]
```diff
@@ -45,39 +40,45 @@ class SharedPreferencesStore implements KeyValueStore {
...
-        _areTypesEqual<T, double?>() ||
-        _areTypesEqual<T, bool>() ||
-        _areTypesEqual<T, bool?>() ||
-        _areTypesEqual<T, String>() ||
-        _areTypesEqual<T, String?>() ||
+    if (_isInt<T>() ||
+        _isDouble<T>() ||
+        _isBool<T>() ||
+        _isString<T>()||
         _areTypesEqual<T, List<String>>() ||
```
Comment (solid-software) [inline lib/src/key_value_store/shared_preferences_store.dart:75]:
```suggestion
        _isListOfStrings<T>() ||
```
to match, what do you think?

File: lib/src/key_value_store/store_error.dart [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
+/// or restoring it from persistence.
 ///
 /// This is probably a configuration error -- check the `KeyValueStore`
 /// implementation and `HydratedSubject` `persist` and `hydrate` callbacks
 /// for type compatibility.
 class StoreError extends Error {
   /// A description of an error.
   final String message;
-
+
```
Comment (solid-software) [inline lib/src/key_value_store/store_error.dart:10]:
```suggestion

```
probably should be removed?

File: test/shared_preferences_store_test.dart [hunk 4]
```diff
@@ -21,98 +19,123 @@ void main() {
...
+          () => const SharedPreferencesStore().get<Exception>('key'),
           throwsA(isA<StoreError>()),
         );
       });
 
       test(
-          'when SharedPreferences return an unsupported type, it throws a StoreError',
-          () {
+          'when SharedPreferences return an unsupported type, '
+          'it throws a StoreError', () {
```
Comment (solid-software) [inline test/shared_preferences_store_test.dart:52]: This looks as interesting topic to discuss on whether we want to go that route

**Review by ghost — CHANGES_REQUESTED** [review]
Looks great - thanks @solid-yuriiprykhodko ! Added a few suggestion, let me know what you think. Overall very pleased with the changes made!

**Review by ghost — APPROVED** [review]
LGTM - great job @solid-yuriiprykhodko !

**Pull request merged**